lightly/utils/scheduler.py

```diff
@@ -37,6 +45,11 @@ def cosine_schedule(
     if max_steps == 1:
         # Avoid division by zero
         decay = end_value
+    elif period:  # "cycle" based on period, if provided
```

Comment:
```suggestion
    if period is not None:  # "cycle" based on period, if provided
```

You could even move this above the `if max_steps == 1:` check